### PR TITLE
Function to validate polis pop data

### PR DESCRIPTION
**Pull request by philbam**
**Summary:** This PR focuses on validation of POLIS population data and introduces a draft, stand-alone quality check function with an explicitly declared `spatial_scale`. It will later be integrated to ingest `spatial_scale` metadata from `load_polis_pop`.

**Notes:**
- Can be tested with population data from EDAV, local file path, or draft `load_polis_pop` (`spatial_scale` declared again for now).
- Reports whether all checks passed or not all checks passed.
- Allows interactive viewing of validation issue details using all, a specific issue number, or none.

#### Unit test request to @chadhunt1 
Hi Chad, could you please help unit test this draft in parallel using mock data of pop.rds? Any edge cases or issues you find would be helpful to revise and finalize. Thanks!

### CODE REVIEW

**Comment by philbam** [comment]
Refactored `check_pop_data_quality` function to use an assertr validation chain with predicate generators for complex rules. The function now returns a table of flagged records with validation columns joined to the original data, replacing the previous interactive/message‑based approach.

#### Unit test request to @chadhunt2 
Hi Chad, you can please proceed with the unit tests. Thank you!

File: R/check_pop_data_quality.R [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+check_pop_data_quality <- function(pop.rds, spatial_scale) {
+  guid_col <- dplyr::case_when(
+    spatial_scale == "dist" ~ "Admin2GUID",
+    spatial_scale == "prov" ~ "Admin1GUID",
+    spatial_scale == "ctry" ~ "Admin0GUID"
+  )
+
+  # Compute missing age groups per GUID (for detail column later)
+  missing_agegroups_df <- pop.rds |>
+    filter(!is.na(.data[[guid_col]])) |>
```
Comment (mcuadera) [inline R/check_pop_data_quality.R]: ensure that when using functions from external packages (dplyr, assertr, etc...), reference the package via `::`. For example:
`dplyr::filter`

In addition add the function as an export by either `usethis::use_package()` or `usethis::use_import_from()`. I recommend importing dplyr in full via `usethis::use_package("dplyr")`. For other functions in a package, please use `usethis::use_import_from("pkg_name", "func_name")`

File: R/check_pop_data_quality.R [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  flagged_cols <- val_cols[sapply(val_cols, function(c) any(result[[c]] == "FLAG", na.rm = TRUE))]
+
+  # Add detail columns if they exist
+  detail_cols <- intersect(c("missing_agegroups", "duplicated_parents"), names(result))
+
+  result <- result |>
+    filter(if_any(all_of(flagged_cols), ~ . == "FLAG")) |>
+    select(row_num, all_of(names(pop.rds)), all_of(flagged_cols), all_of(detail_cols))
+
+  message(sprintf("%d rows flagged with validation flags", nrow(result)))
```
[the remark this answers is not in the document]
Comment (philbam) [reply R/check_pop_data_quality.R]: Modified accordingly.

**Comment by philbam** [comment]
Refactored to work across all spatial scales (`ctry`, `prov`, `dist`) by using `any_of()` for columns that may not exist. Removed `year/startdate mismatch` validation. Updated outlier detection to run separately per age group (`0-5Y`, `0-15Y`, `ALL`) with results in three columns.

**Review by mcuadera — APPROVED** [review]
Given behind schedule, will proceed with merging the function to main without any unit tests...
